### ISSUE TITLE
Extract data on EPUB fallback mechanisms

### ISSUE THREAD

**Issue by rdeltour** [daisy/ace#76]
From @mattgarrish

> It  would also help to know if the publication is using any epub fallbacks,  as they are not accessibility supported and need to be reviewed:
> - fallback attribute on manifest items
> - `bindings` element in the package doc
> - `epub:switch` in the content
> - `epub:trigger` in the content
>
> Not necessarily harmful, but more often than not harmful. What I'm doing is reporting their presence with a warning that they need to be checked to ensure that they aren't being used in a way that accessibility depends on them: using accessible alternatives as fallbacks for spine and switch, disabling default controls after trigger support is established, etc.

**Comment by clapierre** [comment]
In EPUB 3.1 epub:switch and epub:trigger were removed so if these appear in EPUB 3.1 files they would be violations, now we will need to handle these in previous versions of EPUB.  Not sure if you were thinking at this stage with ACE to always just handle these and let EPUBCheck deal with if these should even exist depending on the version of EPUB.

**Comment by rdeltour** [comment]
> Not sure if you were thinking at this stage with ACE to always just handle these and let EPUBCheck deal with if these should even exist depending on the version of EPUB.

Yeah, that would be the idea: Ace doesn't report validation errors, just accessibility violations. So it would be EpubCheck's responsibility to flag them as error in EPUB 3.1, and Ace would just extract the data if present.
Ace currently runs OK in several otherwise invalid EPUBs.

**Comment by clapierre** [comment]
Agreed as long as there is a clear line of understanding what ACE does and does not do we should be fine.
Thanks

**Comment by mattgarrish** [comment]
It would be interesting if epubcheck could be integrated with ace so you didn't have to run them separately (e.g., a separate error section that shows epubchecks output). It's a missing piece in knowing whether 4.1.1 succeeds or not.

But this is going off topic into feature requests.
